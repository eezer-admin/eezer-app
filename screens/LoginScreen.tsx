import * as React from 'react';
import { useContext } from 'react';
import { View, Text, TouchableOpacity, TextInput } from 'react-native';

import Logo from '../components/Logo';
import { AuthContext } from '../contexts/authContext';
import { __ } from '../localization/Localization';
import Styles from '../styles/Styles';

export default function LoginScreen() {
  const [username, onChangeUsername] = React.useState('');
  const [password, onChangePassword] = React.useState('');

  const auth = useContext(AuthContext);

  return (
    <View style={Styles.container}>
      <Logo />

      <TextInput
        onChangeText={onChangeUsername}
        value={username}
        placeholder={__('Username')}
        keyboardType="email-address"
        textContentType="emailAddress"
        style={{ ...Styles.input, marginBottom: Styles.margins.medium }}
      />

      <TextInput
        onChangeText={onChangePassword}
        value={password}
        placeholder={__('Password')}
        keyboardType="default"
        secureTextEntry={true}
        style={{ ...Styles.input, marginBottom: Styles.margins.medium }}
      />

      <TouchableOpacity
        style={{
          ...Styles.button,
          ...Styles.button.green,
        }}
        onPress={() => {
          auth.login(username, password).catch((error) => {
            console.log('Login failed!', error);
          });
        }}>
        <Text style={Styles.button.text}>{__('Log in')}</Text>
      </TouchableOpacity>
    </View>
  );
}
