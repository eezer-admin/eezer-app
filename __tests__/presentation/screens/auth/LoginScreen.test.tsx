import { describe, expect, it } from '@jest/globals';
import LoginScreen from '@presentation/screens/auth/LoginScreen';
import renderer from 'react-test-renderer';

describe('LoginScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LoginScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
