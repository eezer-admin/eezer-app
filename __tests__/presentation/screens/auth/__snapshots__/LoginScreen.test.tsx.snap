// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 1,
      "justifyContent": "center",
      "justifySelf": "center",
      "paddingHorizontal": 15,
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 60,
        "marginBottom": 50,
        "width": 240,
      }
    }
  />
  <TextInput
    keyboardType="email-address"
    onChangeText={[Function]}
    placeholder="Username"
    placeholderTextColor="gray"
    secureTextEntry={false}
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 8,
          "marginBottom": 10,
          "paddingHorizontal": 25,
          "paddingVertical": 10,
          "textAlign": "center",
          "width": "100%",
        },
        null,
      ]
    }
    textContentType="emailAddress"
    value=""
  />
  <TextInput
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="Password"
    placeholderTextColor="gray"
    secureTextEntry={true}
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 8,
          "marginBottom": 10,
          "paddingHorizontal": 25,
          "paddingVertical": 10,
          "textAlign": "center",
          "width": "100%",
        },
        null,
      ]
    }
    value=""
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#86efac",
        "borderRadius": 8,
        "green": {
          "backgroundColor": "#86efac",
        },
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 25,
        "paddingVertical": 20,
        "red": {
          "backgroundColor": "#fca5a5",
          "text": {
            "color": "#b91c1c",
          },
        },
        "text": {
          "color": "#059669",
          "fontWeight": "bold",
          "textTransform": "uppercase",
        },
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#059669",
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      Log in
    </Text>
  </View>
</View>
`;
